mesonbuild/backend/backends.py

```diff
@@ -763,9 +764,14 @@ def rpaths_for_bundled_shared_libraries(self, target: build.BuildTarget, exclude
                 paths.append(os.path.join(self.build_to_src, rel_to_src))
             else:
                 paths.append(libdir)
+        for i in chain(target.link_targets, target.link_whole_targets):
+            if isinstance(i, build.BuildTarget):
+                paths.extend(self.rpaths_for_bundled_shared_libraries(i, exclude_system))
```

Comment:
Looks like this hunk should be in the next commit?

Reply:
yup, you're right, i split these wrong. New version incoming.